File: ios/Components/Modal/SPAlertViewManager.m
//
//  SPAlertViewManager.m
//  bitportal
//
//  Created by Terence Ge on 2019/8/29.
//  Copyright © 2019 Facebook. All rights reserved.
//
#import "React/RCTBridgeModule.h"

@interface RCT_EXTERN_REMAP_MODULE(SPAlert, SPAlertViewManager, NSObject)

+ (BOOL)requiresMainQueueSetup
{
  return YES;
}

RCT_EXTERN_METHOD(presentDone:(nonnull NSString *)title)
RCT_EXTERN_METHOD(presentHeart:(nonnull NSString *)title)
RCT_EXTERN_METHOD(presentTitle:(nonnull NSString *)title)
RCT_EXTERN_METHOD(presentMessage:(nonnull NSString *)title)

@end

File: ios/components/Modal/SPAlertViewManager.swift
//
//  SPAlertViewManager.swift
//  bitportal
//
//  Created by Terence Ge on 2019/8/29.
//  Copyright © 2019 Facebook. All rights reserved.
//

import Foundation

@objc(SPAlertViewManager)
class SPAlertViewManager: NSObject {
  @objc static func requiresMainQueueSetup() -> Bool {
    return true
  }
  
  @objc func presentDone(_ title: NSString) {
    DispatchQueue.main.async {
      SPAlert.present(title: title as String, preset: SPAlertPreset.done)
    }
  }
  
  @objc func presentHeart(_ title: NSString) {
    DispatchQueue.main.async {
      SPAlert.present(title: title as String, preset: SPAlertPreset.heart)
    }
  }
  
  @objc func presentMessage(_ message: NSString) {
    DispatchQueue.main.async {
      SPAlert.present(message: message as String)
    }
  }
  
  @objc func presentTitle(_ title: NSString) {
    DispatchQueue.main.async {
      SPAlert.present(title: title as String)
    }
  }
}
